### PR TITLE
Run manta-pay randomized tests in a loop 10 times

### LINKED ISSUE #704: Run manta-pay randomized tests in a loop in CI
MantaPay tests are currently randomized to a degree, which is great, however sometimes it could take a few tries to catch bugs, depending on what kind of random values are used in each instance.
That's why we should run these tests say 2 - 5 time, however there's a tradeoff that this will slow down the CI, which should be balanced.
This could be a good first issue for someone as it is just changes in CI and measurements.

### PR DESCRIPTION
**Pull request by ghzlatarev**
Signed-off-by: Georgi Zlatarev <georgi.zlatarev@manta.network>

## Description

closes: #704

* Run the randomized manta-pay tests in a loop a few items to avoid false positives.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Added **one** label out of the `L-` group to this PR
- [x] Added **one or more** labels from the `A-` and `C-` groups to this PR
- [x] Explicitly labelled `A-calamari`, `A-dolphin` and/or `A-manta` if your changes are meant for/impact either of these (CI depends on it)
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

Situational Notes:
- If adding functionality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.
- If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. You can run the `metadata_diff.yml` workflow for help. If this number is updated, then the `spec_version` must also be updated
- Verify benchmarks & weights have been updated for any modified runtime logics

### CODE REVIEW

File: pallets/manta-pay/src/test/payment.rs [hunk 1]
```diff
@@ -313,155 +314,186 @@ fn initialize_test(id: AssetId, value: AssetValue) {
...
-        mint_tokens(
-            asset_id,
-            &value_distribution(5, total_free_supply, &mut rng),
-            &mut rng,
-        );
-    });
+    for _ in 0..RANDOMIZED_TESTS_ITERATIONS {
+        let mut rng = OsRng;
+        new_test_ext().execute_with(|| {
+            let asset_id = rng.gen();
```
Comment (Dengjianping) [inline pallets/manta-pay/src/test/payment.rs:320]: Can be the random number sampled from a range like `(0, u128::MAX)`?

Comment (Dengjianping) [reply pallets/manta-pay/src/test/payment.rs:320]: I mean the `amount` not the asset id.

Comment (ghzlatarev) [reply pallets/manta-pay/src/test/payment.rs:320]: no, asset-id is still u32.

Comment (Dengjianping) [reply pallets/manta-pay/src/test/payment.rs:320]: Some of balances can be from (0, u128::MAX) or (0, u64::MAX), depending on the type of integer.

File: pallets/manta-pay/src/test/payment.rs [hunk 2]
```diff
@@ -313,155 +314,186 @@ fn initialize_test(id: AssetId, value: AssetValue) {
...
-        let asset_id = rng.gen();
-        let total_free_supply = AssetValue(rng.gen());
-        initialize_test(asset_id, total_free_supply + DEFAULT_ASSET_ED);
-        mint_tokens(
-            asset_id,
-            &value_distribution(5, total_free_supply, &mut rng),
-            &mut rng,
-        );
-    });
+    for _ in 0..RANDOMIZED_TESTS_ITERATIONS {
```
Comment (Garandor) [inline pallets/manta-pay/src/test/payment.rs]: nit: start from 1 so even if someone puts RANDOMIZED_TESTS_ITERATIONS: usize = 0;
it'll run at least once

**Review by Garandor — APPROVED** [review]
LGTM

**Review by Garandor — DISMISSED** [review]
Actually, fix linter

**Comment by ghzlatarev** [comment]
> Actually, fix linter

yeah im waiting for #781 where I fixed the linter.

**Review by Garandor — APPROVED** [review]
LGTM!

**Review by Dengjianping — APPROVED** [review]

**Pull request merged**